### PR TITLE
B-343: fix ignored user/group renaming

### LINKED ISSUE #343: `opennebula_user`, `opennebula_group`: Changing name has no effect
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

terraform: 1.2.9
provider: all releases

### Affected Resource(s)

- opennebula_user
- openenbula_group

### Expected Behavior

User/group should be recreated or at least an error should be returned

### Actual Behavior

Apply doesn't have any effect and the diff remains across next apply.

### PR DESCRIPTION
**Pull request by treywelsh**
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for PR followers and do not help prioritize the request

### Description

Add ForceNew behavior to the `name` attribute`

### References

Close #343 

### New or Affected Resource(s)

- opennebula_user
- openenbula_group

### Checklist

- [ ] I have created an issue and I have mentioned it in `References`
- [ ] My code follows the style guidelines of this project (use `go fmt`)
- [ ] My changes generate no new warnings or errors
- [ ] I have updated the unit tests and they pass succesfuly
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation (if needed)
- [ ] I have updated the changelog file

### CODE REVIEW

**Comment by treywelsh** [comment]
Changelog is not updated as 1.0.0

**Comment by treywelsh** [comment]
Failed to rename the user:
```
resource "opennebula_group" "group" {
  name                  = "test1"  

  sunstone {
    default_view             = "ig1_customers" 
    group_admin_default_view = "ig1_admin_priv_cloud" 
    group_admin_views        = "ig1_admin_priv_cloud" 
    views                    = "ig1_customers" 
  }

  lifecycle {
    ignore_changes = [
      template,
    ]
  }
}

resource "opennebula_user" "user2" {
  name          = "test5"
  groups = [opennebula_group.group.id]
  password      = "testpwd"
  auth_driver   = "core"
  primary_group = opennebula_group.group.id
}

resource "opennebula_group_admins" "example" {
  group_id = opennebula_group.group.id
  users_ids = [
    opennebula_user.user2.id
  ]
}
```

Apply step:
```
Terraform will perform the following actions:

  # opennebula_group_admins.example will be updated in-place
  ~ resource "opennebula_group_admins" "example" {
        id        = "191"
      ~ users_ids = [
          - 110,
        ] -> (known after apply)
        # (1 unchanged attribute hidden)
    }

  # opennebula_user.user2 must be replaced
-/+ resource "opennebula_user" "user2" {
      ~ default_tags  = {} -> (known after apply)
      ~ id            = "110" -> (known after apply)
      ~ name          = "test2" -> "test5" # forces replacement
      ~ tags_all      = {} -> (known after apply)
        # (4 unchanged attributes hidden)
    }

Plan: 1 to add, 1 to change, 1 to destroy.
╷
│ Warning: Deprecated attribute
│ 
│   on terraform.tf line 1360, in resource "opennebula_group" "group":
│ 1360:       template,
│ 
│ The attribute "template" is deprecated. Refer to the provider documentation for details.
╵

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

opennebula_user.user2: Destroying... [id=110]
opennebula_user.user2: Destruction complete after 0s
opennebula_user.user2: Creating...
opennebula_user.user2: Creation complete after 0s [id=111]
opennebula_group_admins.example: Modifying... [id=191]
╷
│ Error: Failed to delete a group admin
│ 
│   with opennebula_group_admins.example,
│   on terraform.tf line 1381, in resource "opennebula_group_admins" "example":
│ 1381: resource "opennebula_group_admins" "example" {
│ 
│ group (ID: 191): OpenNebula error [NO_EXISTS]: [one.group.deladmin] Error getting user [110].
```

It seems that the group admin resource try to delete the user from it's list but failed because the user doen't exists anymore

**Comment by treywelsh** [comment]
To sum up what's going on:
Rename an user or a group is not possible directly, so some `ForceNew` behavior are added on the `name` of these resources.
During testing I saw a bunch of errors due to the user/group replacement so I added some fixes around:
- empty a group from it's users before deleting it: if it's the primary group of a user, change it for `users` default group, else just delete the group from the user group list.
- add presence checks to user and group before making API calls on old IDs (when they are replaced)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+# 1.1.0 (unreleased)
```
Comment (frousselet) [inline CHANGELOG.md]: 1.0.1

**Comment by treywelsh** [comment]
TODO: add acceptance tests

**Review by frousselet — APPROVED** [review]

**Pull request merged**